substrait/common/StringUtils.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <string_view>
+
+namespace io::substrait::common {
+class StringUtils final {
+ public:
+  static std::string ltrim(std::string_view s) {
+    size_t start = s.find_first_not_of(WHITESPACE);
+    std::string_view res = (start == std::string::npos) ? "" : s.substr(start);
+    return static_cast<std::string>(res);
+  }
+
+  static std::string rtrim(std::string_view s) {
+    size_t end = s.find_last_not_of(WHITESPACE);
+    std::string_view res = (end == std::string::npos) ? "" : s.substr(0, end + 1);
+    return static_cast<std::string>(res);
+  }
+
+  static std::string trim(std::string_view s) {
```

Comment:
These methods should return `std::string_view`.  That way, you leave it up to the caller whether you materialize or not.  Also, you can then do `rtrim(ltrim(s))` without making a redundant temporary copy.